homeassistant/components/igloohome/config_flow.py

```diff
@@ -0,0 +1,60 @@
+"""Config flow for igloohome integration."""
+
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from aiohttp import ClientError
+from igloohome_api import Auth as IgloohomeAuth, AuthException
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigFlow, ConfigFlowResult
+from homeassistant.const import CONF_CLIENT_ID, CONF_CLIENT_SECRET
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_CLIENT_ID): str,
+        vol.Required(CONF_CLIENT_SECRET): str,
+    }
+)
+
+
+class MyConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for igloohome."""
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle the config flow step."""
+
+        errors: dict[str, str] = {}
+        if user_input is not None:
+            auth = IgloohomeAuth(
+                session=async_get_clientsession(self.hass),
+                client_id=user_input[CONF_CLIENT_ID],
+                client_secret=user_input[CONF_CLIENT_SECRET],
+            )
+            try:
+                await auth.async_get_access_token()
+            except AuthException:
+                errors["base"] = "invalid_auth"
+            except ClientError:
+                errors["base"] = "cannot_connect"
+            else:
+                await self.async_set_unique_id(
+                    f"{user_input[CONF_CLIENT_ID]}_{user_input[CONF_CLIENT_SECRET]}"
```

Comment:
with those two you could check earlier. But I don't think that's a good unique_id. This would probably allow setting the same hub (account?) multiple times. Is there something you can get from the account you can use for uniqueness?

Reply:
Hey @zweckj , unfortunately, the `userinfo` endpoint for OAuth2 is also not available with the client credentials. So this is the best option I can find at the moment.

Reply:
are the devices connected to a Hub you can maybe get a serial, etc. from?

Reply:
@zweckj , I've switched over to rejecting the user input based on matching user input. 

To respond to your question, I unfortunately can't see serial numbers being an effective provider of uniqueness. The config entry is tied to a user's account, therefore the number of devices can change and so can the number of hubs which only effectively acts as a bluetooth bridge.